eth/filters/api.go

```diff
@@ -219,11 +222,14 @@ func (s *PublicFilterAPI) newLogFilter(earliest, latest int64, addresses []commo
 	filter.SetAddresses(addresses)
 	filter.SetTopics(topics)
 	filter.LogCallback = func(log *vm.Log, removed bool) {
-		s.logMu.Lock()
-		defer s.logMu.Unlock()
-
-		if queue := s.logQueue[id]; queue != nil {
-			queue.add(vmlog{log, removed})
+		if emit == nil { // queue logs for next poll
+			s.logMu.Lock()
+			defer s.logMu.Unlock()
+			if queue := s.logQueue[id]; queue != nil {
+				queue.add(vmlog{log, removed})
+			}
+		} else {
+			emit <- &event.Event{Time: time.Now(), Data: vmlog{log, removed}}
```

Comment:
I think there's a possibility this will block on slow clients and specifically [this](https://github.com/ethereum/go-ethereum/blob/develop/core/blockchain.go#L1099) part of the code.

We should probably add a timeout mechanism here such like:

```go
select {
case emit <- &event.Event{Time: time.Now(), Data: vmlog{log, removed}}:
case <-time.After(500 * time.Millisecond):
}
```

Or

```go
// emit == make(chan event.Event, buffSize)
} else {
    // drop message if the buffer is full
    if len(emit) == cap(emit) {
        return
    }
    emit <- &event.Event{Time: time.Now(), Data: vmlog{log, removed}}
}
}
```